lib/redis_client/cluster/node.rb

```diff
@@ -92,119 +91,19 @@ def build_connection_prelude
-              a << e.id << e.node_key << e.role << e.primary_id << e.config_epoch
-            end
-          end
-
-          grouped.max_by { |_, v| v.size }[1].first.freeze
-        end
-
-        private
-
-        # @see https://redis.io/commands/cluster-nodes/
-        # @see https://github.com/redis/redis/blob/78960ad57b8a5e6af743d789ed8fd767e37d42b8/src/cluster.c#L4660-L4683
-        def parse_cluster_node_reply(reply) # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
-          reply.each_line("\n", chomp: true).filter_map do |line|
-            fields = line.split
-            flags = fields[2].split(',')
-            next unless fields[7] == 'connected' && (flags & DEAD_FLAGS).empty?
-
-            slots = if fields[8].nil?
-                      EMPTY_ARRAY
-                    else
-                      fields[8..].reject { |str| str.start_with?('[') }
-                                 .map { |str| str.split('-').map { |s| Integer(s) } }
-                                 .map { |a| a.size == 1 ? a << a.first : a }
-                                 .map(&:sort)
-                    end
-
-            ::RedisClient::Cluster::Node::Info.new(
-              id: fields[0],
-              node_key: parse_node_key(fields[1]),
-              role: (flags & ROLE_FLAGS).first,
-              primary_id: fields[3],
-              ping_sent: fields[4],
-              pong_recv: fields[5],
-              config_epoch: fields[6],
-              link_state: fields[7],
-              slots: slots
-            )
-          end
-        end
-
-        # As redirection node_key is dependent on `cluster-preferred-endpoint-type` config,
-        # node_key should use hostname if present in CLUSTER NODES output.
-        #
-        # See https://redis.io/commands/cluster-nodes/ for details on the output format.
-        # node_address matches fhe format: <ip:port@cport[,hostname[,auxiliary_field=value]*]>
-        def parse_node_key(node_address)
-          ip_chunk, hostname, _auxiliaries = node_address.split(',')
-          ip_port_string = ip_chunk.split('@').first
-          return ip_port_string if hostname.nil? || hostname.empty?
-
-          port = ip_port_string.split(':')[1]
-          "#{hostname}:#{port}"
-        end
-      end
-
       def initialize(
-        options,
         concurrent_worker,
-        node_info_list: [],
-        with_replica: false,
-        replica_affinity: :random,
+        config:,
         pool: nil,
         **kwargs
       )
 
         @concurrent_worker = concurrent_worker
-        @slots = build_slot_node_mappings(node_info_list)
-        @replications = build_replication_mappings(node_info_list)
-        klass = make_topology_class(with_replica, replica_affinity)
-        @topology = klass.new(@replications, options, pool, @concurrent_worker, **kwargs)
+        @slots = build_slot_node_mappings([])
+        @replications = build_replication_mappings([])
```

Comment:
Are these initializations needed? It seems to me that these allocations with empty arrays seem to be superfluous.

Reply:
Same as below - I want to set all the ivars in `#initialize` to a valid value, since Ruby recommends this pattern now.